/*
 * This file was automatically generated by EvoSuite
 * Fri Apr 07 20:22:21 KST 2017
 */

package org.prevayler.foundation.network;

import static org.evosuite.runtime.EvoAssertions.assertThrownBy;
import org.junit.AfterClass;
import org.junit.BeforeClass;
import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertTrue;
import static org.junit.Assert.fail;

import java.io.IOException;

import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.Test;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(useVNET = true) 
public class BaseNetworkImpl_ESTest2 extends BaseNetworkImpl_ESTest_scaffolding2 {
@BeforeClass
public static void initEvoSuiteFramework() {
String a = "aa";
}
@AfterClass
public static void clearEvoSuiteFramework() {
String a = "aa";
}

  @Test
  public void test00()  throws Throwable  {
      StubbornNetworkImpl stubbornNetworkImpl0 = new StubbornNetworkImpl();
      StubbornServerReceiverImpl stubbornServerReceiverImpl0 = (StubbornServerReceiverImpl)stubbornNetworkImpl0.newReceiver((Service) null, (ObjectSocket) null);
      assertNotNull(stubbornServerReceiverImpl0);
  }

  @Test
  public void test01()  throws Throwable  {
      StubbornNetworkImpl stubbornNetworkImpl0 = new StubbornNetworkImpl();
      StubbornClientReceiverImpl stubbornClientReceiverImpl0 = (StubbornClientReceiverImpl)stubbornNetworkImpl0.newReceiver("", 1, (ObjectReceiver) null);
      assertTrue(stubbornClientReceiverImpl0.isDaemon());
  }

  @Test
  public void test02()  throws Throwable  {
      StubbornNetworkImpl stubbornNetworkImpl0 = new StubbornNetworkImpl();
      StubbornNetworkProxy stubbornNetworkProxy0 = new StubbornNetworkProxy();
      StubbornClientReceiverImpl stubbornClientReceiverImpl0 = (StubbornClientReceiverImpl)stubbornNetworkImpl0.findServer("M.3BB-.X@8NR_20kOGk", (-356), stubbornNetworkProxy0);
      assertTrue(stubbornClientReceiverImpl0.isDaemon());
  }

  @Test
  public void test03()  throws Throwable  {
      NetworkImpl networkImpl0 = new NetworkImpl();
      // Undeclared exception!
      try { 
        networkImpl0.startService((Service) null, (-1868));
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Port value out of range: -1868
         //
         assertThrownBy("org.evosuite.runtime.mock.java.net.MockServerSocket", e);
      }
  }

  @Test
  public void test04()  throws Throwable  {
      NetworkImpl networkImpl0 = new NetworkImpl();
      // Undeclared exception!
      try { 
        networkImpl0.newReceiver((Service) null, (ObjectSocket) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("org.prevayler.foundation.network.NetworkClientObjectReceiverImpl", e);
      }
  }

  @Test
  public void test05()  throws Throwable  {
      NetworkImpl networkImpl0 = new NetworkImpl();
      // Undeclared exception!
      try { 
        networkImpl0.newReceiver((String) null, (-1), (ObjectReceiver) null);
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // port out of range:-1
         //
         assertThrownBy("java.net.InetSocketAddress", e);
      }
  }

  @Test
  public void test08()  throws Throwable  {
      StubbornNetworkImpl stubbornNetworkImpl0 = new StubbornNetworkImpl();
      // Undeclared exception!
      try { 
        stubbornNetworkImpl0.createService((Service) null, (-1));
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Port value out of range: -1
         //
         assertThrownBy("org.evosuite.runtime.mock.java.net.MockServerSocket", e);
      }
  }

  @Test
  public void test11()  throws Throwable  {
      NetworkImpl networkImpl0 = new NetworkImpl();
      networkImpl0.startService((Service) null, 1733);
      networkImpl0.stopService(1733);
  }

  @Test
  public void test12()  throws Throwable  {
      NetworkImpl networkImpl0 = new NetworkImpl();
      try { 
        networkImpl0.stopService(1733);
        fail("Expecting exception: IOException");
      
      } catch(IOException e) {
         //
         // Port Not in Service
         //
         assertThrownBy("org.prevayler.foundation.network.BaseNetworkImpl", e);
      }
  }

  @Test
  public void test13()  throws Throwable  {
      NetworkImpl networkImpl0 = new NetworkImpl();
      networkImpl0.startService((Service) null, 0);
      try { 
        networkImpl0.startService((Service) null, 0);
        fail("Expecting exception: IOException");
      
      } catch(IOException e) {
         //
         // Port In Service
         //
         assertThrownBy("org.prevayler.foundation.network.BaseNetworkImpl", e);
      }
  }
}
